packages/react-router/src/client/ReactRouterClerkProvider.tsx

```diff
@@ -119,9 +119,9 @@ type ClerkReactRouterOptions = Partial<
   Omit<ReactRouterClerkProviderProps, 'routerPush' | 'routerReplace' | 'clerkState'>
 >;
 
-// TODO: Remove "any" on loaderData type
+// TODO: Remove "any" on loaderData type and use Route.ComponentProps from userland code
 type ClerkProviderProps = ClerkReactRouterOptions & {
-  loaderData?: ReactRouterComponentProps['loaderData'];
+  loaderData?: any;
```

Comment:
This was already resolving to `any` previously, so we can remove the reliance on `react-router/internal` here